File: cpp/05/pe5.cpp
#include <iostream>
#include <stdio.h>
#include <math.h>

bool is_prime(unsigned int num);

int main(){
    system("CLS");
    unsigned long int result = 1;
    int temp, MAX_VAL = 20;
    for(int i=2; i<=MAX_VAL; i++){
        if(is_prime(i)){
          temp = i;
          while((temp*i)<MAX_VAL) temp *= i;
           result *= temp;
        }
    }
    std::cout<<result<<std::endl;
    return 0;
}

bool is_prime(unsigned int num){
    if(num == 2) return true;
    else if(!(num%2)) return false;
    else{
        for(unsigned int i=3; i<=(sqrt(num + 1)); i+=2){
            if(!(num%i)) return false;
        }
        return true;
    }
}

File: python/pe5.py
from math import sqrt
from os import system 
system('cls')

def is_prime(num):
    if num == 2: return True
    elif not(num%2): return False
    else:
        for i in range(3, int(sqrt(num)) + 1, 2):
            if(not(num%i)): return False
        return True

if __name__ == '__main__':
    MAX_VAL = 20
    result = 1
    for i in range(2, MAX_VAL + 1):
        if is_prime(i):
            temp = i
            while (temp*i) < MAX_VAL: temp *= i
            result *= temp
    print(result)